### ISSUE TITLE
Async global setup doesn't work properly 

### ISSUE THREAD

**Issue by m3hari** [lukeed/uvu#127]
First of all, Thank you very much for the amazing package ❤️!

## Context
I want to run a global setup script before starting tests. 
My actual use case is starting `playwright browser server` and making a `ws` connection on each test suite to open a tab.

## What I did

I have a typescript file that does some setup tasks in `scripts/test-setup.ts` and I have the following command to 
run tests
`   
 "test": "uvu -r ts-node/register -r scripts/test-setup tests"
`

## What happened
`uvu` starts running suits before the `setup` script finishes  

## Expectation

Run tests after the global setup script exits  

Here is a small repo to reproduce the issue   https://github.com/m3hari/uvu-issue-sandbox

![image](https://user-images.githubusercontent.com/9920926/127762746-ef8af869-0b05-43d8-bf6a-5a0fb1b29c81.png)

**Comment by lukeed** [comment]
This type of setup file needs top-level await in order for it to finish executing before the file "exits". Unfortunately, TLA is only available within ESM files and any `--require/-r` usage expects a CommonJS file.

So, in order to achieve this, this type of feature requires either #113 or #81.

Both of these will solve this through separate avenues. Closing this in favor of them.
Thanks!
